toolchain/check/deduce.cpp

```diff
@@ -297,7 +298,8 @@ auto DeductionContext::Deduce() -> bool {
     // TODO: Bail out if there's nothing to deduce: if we're not in a pattern
     // and the parameter doesn't have a symbolic constant value.
 
-    auto param_type_id = context().insts().Get(param_id).type_id();
+    auto param_type_id = ExtractScrutineeType(
+        context().types(), context().insts().Get(param_id).type_id());
```

Comment:
Answering my own question in ExtractScrutineeType, the inst here is not necessarily a Pattern at all. But then maybe it could be more like:
```suggestion
    auto param_type_id = context().insts().Get(param_id).type_id();
    if (context.types().Is<SemIR::PatternType>(param_type_id)) {
      param_type_id = ExtractScrutineeType(
        context().types(), param_type_id);
    }
```
This would let you see the flow directly in the debugger while stepping over stuff, and we could narrow ExtractScrutineeType() down to just PatternType?